osidb/sync_manager.py

```diff
@@ -85,9 +99,25 @@ def schedule(cls, sync_id, *args, schedule_options=None, **kwargs):
         if schedule_options is None:
             schedule_options = {}
 
-        cls.objects.get_or_create(sync_id=sync_id)
+        created = cls.objects.get_or_create(sync_id=sync_id)[1]
         cls.objects.filter(sync_id=sync_id).update(last_scheduled_dt=timezone.now())
 
+        if not created and cls.MODE == cls.SyncManagerMode.EXCLUSIVE:
+            # If the task is already scheduled, skip
+            if not cls.is_scheduled(sync_id):
```

Comment:
I am not sure I understand this part, the comment says that we skip when the task is already scheduled but the condition is reversed I think, we perform the skip if it is not scheduled? :thinking: So currently we would perform reschedule only if the task is not already scheduled, which I think handles the latter part of the `schedule` method.

Reply:
Sorry the comment is a bit misleading, is referring that we skip the if itself 😆

The idea is that this `is_scheduled` logic is only used for knowing when we need to schedule a task to run in the future.

Reply:
Clarified in a meeting, looking good. :+1: